### PR TITLE
BR agent sidecar support

### PR DESCRIPTION
**Pull request by hasancelik**

### CODE REVIEW

**Review by dzeromski-hazelcast — APPROVED** [review]

File: controllers/hazelcast/br_agent_rest_client.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	for _, address := range ac.addresses {
+		req, err := postRequestWithBody(ctxT, reqBody, address, uploadBackup)
+		if err != nil {
+			return fmt.Errorf("request creation failed: %s, address --> %q , URL --> %q ", err, address, address+uploadBackup)
+		}
+		res, err := ac.executeRequest(req)
+		if err != nil {
+			return err
+		}
+		defer res.Body.Close()
```
Comment (cagric0) [inline controllers/hazelcast/br_agent_rest_client.go:63]: Do we need to call `Close` function? I think we don't need since we don't open(read) `res.Body`.

**Review by cagric0 — APPROVED** [review]
I have two little question. Other than that LGTM!

File: api/v1alpha1/hotbackup_types.go [hunk 2]
```diff
@@ -38,7 +38,11 @@ type HotBackupSpec struct {
 	//	@daily (or @midnight)  | Run once a day, midnight                   | 0 0 * * *
 	//	@hourly                | Run once an hour, beginning of hour        | 0 * * * *
 	// +optional
-	Schedule string `json:"schedule,omitempty"`
+	Schedule string `json:"schedule"`
+
+	// URL of the bucket to download HotBackup folders.
+	// +optional
+	BucketURL string `json:"bucket"`
```
[the remark this answers is not in the document]
Comment (hasancelik) [reply api/v1alpha1/hotbackup_types.go:45]: Users can create HotBackup CR without external storage

File: controllers/hazelcast/br_agent_rest_client.go (hunk 1 above)
Comment (hasancelik) [reply controllers/hazelcast/br_agent_rest_client.go:63]: You do not need to open the body, it is used by the request itself.

**Pull request merged**